lib/babe/babe.go

```diff
@@ -336,158 +303,105 @@ func (b *Service) initiate() {
 		return
 	}
 
-	err := b.invokeBlockAuthoring()
-	if err != nil {
-		logger.Criticalf("block authoring error: %s", err)
+	// we should consider better error handling for this - we should
+	// retry to run the engine at some point (maybe the next epoch) if
+	// there's an error.
+	if err := b.runEngine(); err != nil {
+		logger.Criticalf("failed to run block production engine: %s", err)
 	}
 }
 
-func (b *Service) invokeBlockAuthoring() error {
-	epoch, err := b.epochState.GetCurrentEpoch()
+func (b *Service) initiateAndGetEpochHandler(epoch uint64) (*epochHandler, error) {
```

Comment:
Please use named returns 😃 

```suggestion
func (b *Service) initiateAndGetEpochHandler(epoch uint64) (epochHandler *epochHandler, err error) {
```

Reply:
why here? the variable name is the same as the type name, not sure if it adds anything